lua/autorun/player_extension.lua

```diff
@@ -155,4 +155,111 @@ function PLAYER:UsingWeapon(cls)
     local c = self:GetActiveWeapon()
 
     return IsValid(c) and c:GetClass() == cls
+end
+
+local function AppropriateNavArea(ply, area)
+    local mins, maxs = ply:GetCollisionBounds()
+    if (area:GetSizeX() < maxs.x * 2 or area:GetSizeY() < maxs.z * 2) then return false end
+
+    local mins,maxs = ply:GetCollisionBounds()
+    local tr = {}
+    tr.start = area:GetCenter() + Vector(0,0,1)
+    tr.endpos =area:GetCenter()
+    tr.filter = function(ent)
+        if(ent == ply)then return false end
+        local should = hook.Call("ShouldCollide",nil,ply,ent) or true
+        return should
+    end
+    tr.mins = mins
+    tr.maxs = maxs
+    local trace = util.TraceHull(tr)
+    if(trace.StartSolid)then return false end
+    
+    return true
+end
+
+-- multiply disance along an axis, so we can  
+local function biasdist(pos1, pos2, biasvec)
+    return (pos1 * biasvec):Distance((pos2 * biasvec))
+end
+
+function PLAYER:IsStuck()
+    if(self:InVehicle())then return false end
+    local mins,maxs = self:GetCollisionBounds()
+    local tr = {}
+    tr.start = self:GetPos()
+    tr.endpos = tr.start + self:GetVelocity()*FrameTime()
+    tr.filter = function(ent)
+        if(ent == self)then return false end
+        local should = hook.Call("ShouldCollide",nil,self,ent) or true
+        return should
+    end
+    tr.mins = mins
+    tr.maxs = maxs
+    local trace = util.TraceHull(tr)
+    if(trace.StartSolid)then return true end
+    return false
+end
+
+if (SERVER) then
+    concommand.Add("stuck", function(ply, cmd, args)
```

Comment:
What if this gets spammed?

Reply:
The command is throttled, but I could probably optimize this further

Reply:
wait no that's not supposed to still be there

Reply:
i mostly mean make sure there is a rate limit on it (10 sec cooldown)

Reply:
well i use whatever throttling fedorachat has. it also refuses to do any of the unstick functions if it doesn't think you're stuck